### PR TITLE
Adding FS TC_83583723

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding FS TC_83583723
CEPH-83583723 - Ensure deletion of clones are allowed when the clones are stuck and in pending state
Test Steps:
    Created 7 clones in which 3 will be going to pending state and 4 in-progess state.
    we are waiting for all clones to get created and deleted them and recreating them in for loop
        for i in {1..10};do echo $i; for i in {1..7};do echo $i;
        ceph fs subvolume snapshot clone cephfs subvol_2 snap_1 clone_${i};
        echo "Creation of clone Done"; done; sh clone_status.sh;for i in {1..7};do echo $i;
        ceph fs subvolume rm cephfs clone_${i};echo "Deletion of volume done"; done;
        echo "##########################################"; done

    In parallel failed mds with ceph mds fail 0 in loop and
    waited for it to become active and waited for 30 seconds this is done for 100 times in loop
        for i in {1..100};do echo $i; ceph mds fail 0; sh test_mds.sh;
        echo "##########################################"; done

Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-63AOGQ/        

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_bugs/test_clone_delete_with_FS_fail.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                "target_subvol_name": f"clone_{x}",
+                "group_name": "subvolgroup_clone_cancel_1",
+            }
+            for x in range(1, 8)
+        ]
+        rmclone_list = [
+            {"vol_name": default_fs, "subvol_name": f"clone_{x}"} for x in range(1, 8)
+        ]
+        for i in range(1, 10):
+            log.info(f"Iteration {i} ")
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_bugs/test_clone_delete_with_FS_fail.py:128]: I believe in the interest of time, the logs were captured for 4 iterations. please ack if that is the case and we are good to run for 10 iterations.

**Review by hkadam134 — DISMISSED** [review]
lgtm.

File: tests/cephfs/cephfs_bugs/test_clone_delete_with_FS_fail.py (hunk 1 above)
Comment (AmarnatReddy) [reply tests/cephfs/cephfs_bugs/test_clone_delete_with_FS_fail.py:128]: yes Neha

File: tests/cephfs/cephfs_bugs/test_clone_delete_with_FS_fail.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        return 0
+
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Clean Up in progess")
+        rmclone_list = [
+            {"vol_name": default_fs, "subvol_name": "clone_status_3", "force": True},
```
[the remark this answers is not in the document]
Comment (AmarnatReddy) [reply tests/cephfs/cephfs_bugs/test_clone_delete_with_FS_fail.py]: We don't need this. 
Removed those lines of code

**Review by hkadam134 — APPROVED** [review]
LGTM

**Review by neha-gangadhar — APPROVED** [review]
Approved

**Pull request merged**